### PR TITLE
[CDF-333]: Added a theme in order to customize index page. Added some ex...

### PR DESCRIPTION
**Pull request by pamval**
...amples and context to the Dashboard classes

@jvelasques @afrjorge review, please

### CODE REVIEW

File: cdf-core/cdf/js-modules/Dashboard.Blueprint.js [hunk 1]
```diff
@@ -13,12 +13,44 @@
...
+
+
+/**
  * Represents a blueprint dashboard. It's a specialization of the base abstract Dashboard class.
  * Require as cdf/Dashboard.Blueprint
  *
+ * Usage example:
+ *
+ *      require(['cdf/Dashboard.Blueprint', 'cdf/components/ButtonComponent'],
+ *        function(Dashboard, ButtonComponent, TextInputComponent) {
```
Comment (afrjorge) [inline cdf-core/cdf/js-modules/Dashboard.Blueprint.js]: TextInputComponent should be removed

File: cdf-core/cdf/js-modules/Dashboard.Bootstrap.js [hunk 2]
```diff
@@ -11,14 +11,47 @@
...
+
 
 /**
  * Represents a bootstrap dashboard. It's a specialization of the base abstract Dashboard class.
  * Require as cdf/Dashboard.Bootstrap
  *
+ * Usage example:
+ *
+ *      require(['cdf/Dashboard.Bootstrap', 'cdf/components/ButtonComponent'],
+ *        function(Dashboard, ButtonComponent, TextInputComponent) {
```
Comment (afrjorge) [inline cdf-core/cdf/js-modules/Dashboard.Bootstrap.js]: TextInputComponent should be removed

File: cdf-core/cdf/js-modules/Dashboard.Clean.js [hunk 3]
```diff
@@ -11,14 +11,47 @@
...
+
 
 /**
  * Represents a clean (no css engine) dashboard. It's a specialization of the base abstract Dashboard class.
  * Require as cdf/Dashboard.Clean
  *
+ * Usage example:
+ *
+ *      require(['cdf/Dashboard.Clean', 'cdf/components/ButtonComponent'],
+ *        function(Dashboard, ButtonComponent, TextInputComponent) {
```
Comment (afrjorge) [inline cdf-core/cdf/js-modules/Dashboard.Clean.js]: TextInputComponent should be removed

**Comment by pamval** [comment]
@afrjorge, changed.

File: cdf-core/cdf/js-modules/components/UnmanagedComponent.js [hunk 4]
```diff
@@ -27,6 +27,199 @@ define(['./BaseComponent', 'amd!../lib/underscore', '../lib/jquery', '../Logger'
...
+   *
+   * <pre><code>preExecution -&gt; update -&gt; postExecution -&gt; query -&gt; postFetch -&gt; redraw
+   * </code></pre>
+   *
+   * <p>Which breaks the contract for postExecution running after the component is done
+   * updating. The solution here is that the component itself must take control of
+   * postExecution, while keeping the burden of implementing the lifecycle in CDF
+   * rather than passing it to the component developer. On a related topic, postFetch
+   * has become a de facto standard part of the lifecycle, yet its implementation was
+   * left to the component implenters, which leads to a fairly large amount of
```
Comment (jvelasques) [inline cdf-core/cdf/js-modules/components/UnmanagedComponent.js]: typo - implementers or maybe even developers?

File: cdf-core/cdf/js-modules/components/UnmanagedComponent.js [hunk 5]
```diff
@@ -27,6 +27,199 @@ define(['./BaseComponent', 'amd!../lib/underscore', '../lib/jquery', '../Logger'
...
+   *
+   * <h2>Component execution order and Priority</h2>
+   *
+   * <p>There are no major changes in the way components behave. There is, however an
+   * important caveat - since all components (that have been converted) will be
+   * executed simultaneously, we can no longer rely on the order of execution. </p>
+   *
+   * <p>There's now an additional property named <em>priority</em>. The priority of component
+   * execution, defaulting to 5. The lower the number, the higher priority the
+   * component has. Components with same priority with be executed simultaneously.
```
Comment (jvelasques) [inline cdf-core/cdf/js-modules/components/UnmanagedComponent.js]: Components with the same priority will be executed simultaneously

File: cdf-core/cdf/js-modules/components/UnmanagedComponent.js [hunk 6]
```diff
@@ -27,6 +27,199 @@ define(['./BaseComponent', 'amd!../lib/underscore', '../lib/jquery', '../Logger'
...
+   * of the Component, called <code>redraw</code>. It's also considered standard practice to
+   * use <code>Function#bind</code> or <code>_.bind</code> to ensure that, inside the <code>redraw</code> callback,
+   * <code>this</code> points to the component itself.</p>
+   *
+   * <h3>Use <code>synchronous</code> If Your Component Doesn't Use External Data</h3>
+   *
+   * <p>Components that don't use any external data at all can continue subclassing
+   * BaseComponent without any change of functionality. However, for the sake of
+   * consistency (or because you want querying to be optional -- see the section for
+   * details), your can use subclass UnmanagedComponent and use the <code>synchronous</code>
```
Comment (jvelasques) [inline cdf-core/cdf/js-modules/components/UnmanagedComponent.js]: typo - you

File: cdf-core/cdf/js-modules/components/UnmanagedComponent.js [hunk 7]
```diff
@@ -27,6 +27,199 @@ define(['./BaseComponent', 'amd!../lib/underscore', '../lib/jquery', '../Logger'
...
+   * rather than passing it to the component developer. On a related topic, postFetch
+   * has become a de facto standard part of the lifecycle, yet its implementation was
+   * left to the component implenters, which leads to a fairly large amount of
+   * boilerplate code.</p>
+   *
+   * <p>Our objective here was to retool the base component so as to deal with both
+   * of these issues, thus allowing queries to be performed asynchronously while
+   * reducing the developer effort involved in creating a component.</p>
+   *
+   * <h2>Component execution order and Priority</h2>
```
Comment (jvelasques) [inline cdf-core/cdf/js-modules/components/UnmanagedComponent.js]: Start every word uppercase for consistensy

File: cdf-core/cdf/js-modules/components/UnmanagedComponent.js [hunk 8]
```diff
@@ -27,6 +27,199 @@ define(['./BaseComponent', 'amd!../lib/underscore', '../lib/jquery', '../Logger'
...
+   * execution, defaulting to 5. The lower the number, the higher priority the
+   * component has. Components with same priority with be executed simultaneously.
+   * Useful in places where we need to give higher priority to filters or other
+   * components that need to be executed before other components.</p>
+   *
+   * <p>This way there's no longer the need to use dummy parameters and postChange
+   * tricks to do, for instance, cascade prompts.</p>
+   *
+   *
+   * <h2>Backward compatibility and changes</h2>
```
Comment (jvelasques) [inline cdf-core/cdf/js-modules/components/UnmanagedComponent.js]: Start every word uppercase for consistensy

**Comment by pamval** [comment]
@jvelasques fixed

**Pull request merged**